### DISCUSSION TITLE
Task deploy:update_code permission denied

### DISCUSSION THREAD

**Discussion by liberatintt** [deployphp/deployer#2448] [General]
- Deployer version: 6.8.0
- Deployment OS: Oracle Linux

**host config:**
host('....')
    ->stage('production')
    ->user('ssh_user')
    ->become('doc_root_allowed_user')
    ->set('branch', 'newversion')
    ->set('deploy_path', '/var/www/html/default');

**Deploy task definition:**

task('deploy', [
    'deploy:info',
    'deploy:prepare',
    'deploy:lock',
    'deploy:release',
    'deploy:update_code',
    'deploy:shared',
    'deploy:writable',
    'deploy:vendors',
    'deploy:clear_paths',
    'deploy:symlink',
    'deploy:unlock',
    'cleanup',
    'success'
]);

At the deploy:update_code task i always get a permission denied (it look like the become setting is not handled by this task)

PS: I am sorry for my poor english

**Comment by antonmedv** [comment]
Looks like problem on your side.

**Reply by liberatintt** [reply]
There is a way to execute the deploy:update_code with a different user?

**Reply by antonmedv** [reply]
You can override any task as you want.

**Reply by liberatintt** [reply]
Thx, it worked for me
